docs/manual/src/udl/structs.md

```diff
@@ -57,9 +57,78 @@ struct TodoEntry {
 }
 ```
 
-Depending on the languge, the foreign-language bindings may also need to be aware of
+Depending on the language, the foreign-language bindings may also need to be aware of
 these embedded references. For example in Kotlin, each Object instance must be explicitly
 destroyed to avoid leaking the underlying memory, and this also applies to Objects stored
 in record fields.
 
 You can read more about managing object references in the section on [interfaces](./interfaces.md).
+
+## Default values for fields
+
+Fields can be specified with a default value:
+
+```idl
+dictionary TodoEntry {
+    boolean done = false;
+    string text;
+};
+```
+
+The corresponding generated Kotlin code would be equivalent to:
+
+```kotlin
+data class TodoEntry (
+    var done: Boolean = false,
+    var text: String
+)  {
+    // ...
+}
+```
+
+This works for Swift and Python targets too.
+If not set otherwise the default value for a field is passed to the Rust constructor.
+
+## Optional fields and default values
+
+Fields can be made optional using a `T?` type.
```

Comment:
I could be convinced that this is not necessary, as the `T?` thing is already document in Types and thus having default values for it with `null` is covered by the above paragraph.
I still figured it's worth calling out specifically here.

Just let me know what you think, I can change that.

Reply:
I don't think it hurts to mention it here too. However, I suggest maybe adding what the rust receiver looks like too (ie, takes an Option<T>), just like you've done above for the defaults.

Reply:
I agree it seems worth calling out here as well, it may be easier to understand with these concrete examples.